crdt.go

```diff
@@ -473,8 +527,18 @@ func (store *Datastore) sendJobWorker() {
 		select {
 		case <-store.ctx.Done():
 			close(store.jobQueue)
+			// drain jobs
+			for j := range store.jobQueue {
+				j.session.Done()
+			}
 			return
 		case j := <-store.sendJobs:
+			// TODO: jobqueue could be buffered. But then we'd
+			// need to make sure workers abort when
+			// store.ctx.Done() even if more jobs are made
+			// available until the other case closes the channel.
+			// Not overly worried, may not make much difference
+			// anyways.
```

Comment:
Why are we not worried? Either the worker is guaranteed to drain the job queue and the code on lines 531-533 isn't necessary or this code can deadlock (when the worker _doesn't_ drain the queue).

Reply:
workers should drain the queue. I've changed that, and buffered the channel.